code/game/objects/items/devices/chameleonproj.dm

```diff
@@ -60,6 +61,15 @@
 	temp.plane = initial(target.plane)
 	saved_appearance = temp.appearance
 
+	if(istype(target, /atom/movable)) //Record vis_overlays data
+		var/atom/movable/M = target
+		var/count = 0
+		var/list/temp_vis_overlay_data = list()
```

Comment:
What's the purpose of the temp list

Reply:
The list is used to record information on the target's current overlay data, so if the object that was scanned has its overlay data changed (for example, of the scanned object is destroyed), the projector uses the record. It works similarly with regular appearance, storing a copy of the target's current appearance to use later.

Reply:
Now that I think about it, there really is no point in having the temporary list instead of just writing to vis_overlay_data directly. I'll get rid of it.